### DISCUSSION TITLE
[Solved] - API query via ANSIBLE under Netbox 4.0+

### DISCUSSION THREAD

**Discussion by LLFT** [netbox-community/netbox#17025] [Help Wanted!] [answered]
Hi,
For several months now, I've been working on a POC (Proof of Concept) on Netbox and its automation via Ansible for the creation of various elements of our infrastructure.
For this, I'm using version 3.7.5 of Netbox.

The POC must move to the testing phase, and for this, I've created a new machine with the same system configuration. But it's more up-to-date with the Netbox version (4.0.7). I've imported my Ansible role to manage our infrastructure.
And when I execute it, I get an error: "Failed to establish connection to NetBox API". Nothing has changed between my POC and my testing phase, except for the TOKEN.

When I try the command: `curl -k -H "Authorization: Token $NETBOX_TOKEN" "${NETBOX_API}/api/status/"` with the appropriate variables, I get the expected result:

`{"django-version":"5.0.7","installed-apps":{"debug_toolbar":"4.3.0","django_filters":"24.2","django_prometheus":"2.3.1","django_rq":"2.10.2","django_tables2":"2.7.0","drf_spectacular":"0.27.2","drf_spectacular_sidecar":"2024.7.1","mptt":"0.16.0","rest_framework":"3.15.2","social_django":"5.4.1","taggit":"5.0.1","timezone_field":"7.0"},"netbox-version":"4.0.7","plugins":{},"python-version":"3.11.2","rq-workers-running":1}`

I'm going in circles trying to find out why I have this error, and I'm not getting anywhere.

Is a minimum amount of information required in Netbox to be able to use Ansible in Create mode? 

The first task of my role is to create a site.
I activated the logs by following this procedure : [HERE](https://majornetwork.net/2022/09/configuring-logging-in-netbox/)

I've seen that the configuration.py has evolved between the two versions.
Have I forgotten something in my configuration?

Thank you for your help.

**Comment by LLFT** [comment]
I just found out that the version of python3-pynetbox on Debian 12 was only at 7.0.0-1, while the issue [#1265](https://github.com/netbox-community/ansible_modules/issues/1265) recommends upgrading to 7.34.
I will look into that direction.

**Comment by LLFT** [comment, marked as answer]
I can confirm that the solution lies in the update of python3-pynetbox.
Since the repositories are not up-to-date, we can proceed as follows:
```
sudo apt remove python3-pynetbox
pip3 install pynetbox==7.3.4 --break-system-packages
```

**Reply by candlerb** [reply]
FWIW, using a virtualenv for your application is in almost all cases better than having conflicting versions of packages, or a mix of system packages and pip globally installed packages.
